### PR TITLE
Omit non referenced expressions

### PR DESCRIPTION
**Pull request by teoxoy**

### CODE REVIEW

**Comment by teoxoy** [comment]
Seems HLSL validation is failing on https://github.com/gfx-rs/naga/issues/2366 which should be fixed by https://github.com/gfx-rs/naga/pull/2367.

File: src/valid/analyzer.rs [hunk 1]
```diff
@@ -675,12 +675,17 @@ impl FunctionInfo {
                 requirements: UniformityRequirements::empty(),
             },
             E::Math {
-                arg, arg1, arg2, ..
+                fun: _,
+                arg,
+                arg1,
+                arg2,
+                arg3,
```
Comment (jimblandy) [inline src/valid/analyzer.rs:682]: Nice catch. I think I've seen several of these fixed recently. I did a quick look and I think this must the last - the other places that use `Math { .. }` are sites that can't produce results at all without the arguments, so `..` is probably okay there.

File: src/valid/analyzer.rs [hunk 2]
```diff
@@ -868,7 +873,7 @@ impl FunctionInfo {
                 S::Loop {
                     ref body,
                     ref continuing,
-                    break_if: _,
+                    break_if,
```
Comment (jimblandy) [inline src/valid/analyzer.rs:876]: Another nice catch.

**Review by jimblandy — APPROVED** [review]
Looks good!

**Pull request merged**